core/src/node_interface.rs

```diff
@@ -62,11 +63,23 @@ impl SignTransactionWithInputs for NodeInterface {
     }
 }
 
-pub fn assert_wallet_unlocked(node: &NodeInterface) {
-    let unlocked = node.wallet_status().unwrap().unlocked;
+pub fn assert_wallet_unlocked(node: &NodeApi) {
```

Comment:
Since this is no longer just an assertion lets rename it to `try_ensure_wallet_unlocked` or similar

Reply:
Adjusted in recent commit.